cicecore/cicedynB/infrastructure/ice_restart_driver.F90

```diff
@@ -358,21 +358,27 @@ subroutine restartfile (ice_ic)
            'vvel',1,diag,field_loc_NEcorner, field_type_vector)
 
       if (grid_ice == 'CD') then
-         call read_restart_field(nu_restart,0,uvelE,'ruf8', &
-              'uvelE',1,diag,field_loc_Eface, field_type_vector)
-         call read_restart_field(nu_restart,0,vvelE,'ruf8', &
-              'vvelE',1,diag,field_loc_Eface, field_type_vector)
-         call read_restart_field(nu_restart,0,uvelN,'ruf8', &
-              'uvelN',1,diag,field_loc_Nface, field_type_vector)
-         call read_restart_field(nu_restart,0,vvelN,'ruf8', &
-              'vvelN',1,diag,field_loc_Nface, field_type_vector)
+         if (query_field(nu_restart,'uvelE')) &
+            call read_restart_field(nu_restart,0,uvelE,'ruf8', &
```

Comment:
Nice.  Do restarts now work for C- and CD-grids?

Reply:
On restart, the C/CD restart tests are aborting.  @dabail10 suggested C/CD restarts are working, so we need to just check that.  The fix here allows the C/CD cases to start from B restarts which is a (temporary?) requirement to get some of the test cases to run.

Reply:
So, these only work for ice_ic = 'default'. I had that in the PR. I had not added the capability for reading the B-grid initial files. This is one piece, which is great, but I was thinking that for a CD or C test reading the B-grid restart that we would initialize the C/CD velocities from the B-grid.

Reply:
Thanks for adding this.

Reply:
The gridsys_suite starts from B restarts.  I agree we could do more to initialize from B, but the C/CD smoke tests from B restarts run fine.  I don't understand why ice_ic='default' plays a role in the restart test.  The C/CD restart test starts from a B restart fine and writes a restart after 5 days (or so).  When the 2nd run of the restart test starts up, it definitely reads the C/CD fields from the new restart file then aborts in the shortwave code somewhere.  I think with these fixes, the ice_ic='default' problem should be fixed.  What restart tests were you running?  Can you review the restart tests in the gridsys_suite and see how they differ from your test when you have a chance.  Thanks.